include/dlaf/eigensolver/reduction_to_band/mc.h

```diff
@@ -300,32 +313,42 @@ void setupReflectorPanelV(bool has_head, const LocalTileSize& ai_offset, const S
   // computations, they should be well-formed, i.e. a unit lower trapezoidal matrix. For this reason,
   // a support tile is used, where just the reflectors values are copied, the diagonal is set to 1
   // and the rest is zeroed out.
-  auto it_begin = v.iteratorLocal().begin();
-  auto it_end = v.iteratorLocal().end();
+  auto it_begin = panel_view.iteratorLocal().begin();
+  auto it_end = panel_view.iteratorLocal().end();
 
   if (has_head) {
     auto setupV0 = pika::unwrapping([](auto&& tile_v, const auto& tile_a) {
       matrix::internal::copy(tile_a, tile_v);
       tile::internal::laset(lapack::MatrixType::Upper, T(0), T(1), tile_v);
     });
 
+    const LocalTileIndex i = *it_begin;
+    matrix::SubTileSpec spec = panel_view(i);
+
     // Note:
     // If the number of reflectors are limited by height (|reflector| > 1), the panel is narrower than
     // the blocksize, leading to just using a part of A (first full nrefls columns)
-    const auto malformed_v0_idx = indexFromOrigin(ai_offset);
-    const auto nrows = mat_a.tileSize(GlobalTileIndex(Coord::Row, v.rangeStart())).rows();
-    const auto& tile_a = splitTile(mat_a.read(malformed_v0_idx), {{0, 0}, {nrows, nrefls}});
+    spec.size = {spec.size.rows(), std::min(nrefls, spec.size.cols())};
 
-    pika::dataflow(getHpExecutor<Backend::MC>(), std::move(setupV0), v(malformed_v0_idx), tile_a);
+    auto tile_v = v(i);
+    pika::dataflow(getHpExecutor<Backend::MC>(), std::move(setupV0), std::move(tile_v),
+                   splitTile(mat_a.read(i), spec));
 
     ++it_begin;
   }
 
   // The rest of the V panel of reflectors can just point to the values in A, since they are
   // well formed in-place.
   for (auto it = it_begin; it < it_end; ++it) {
-    const LocalTileIndex idx(it->row(), ai_offset.cols());
-    v.setTile(idx, mat_a.read(idx));
+    const LocalTileIndex idx = *it;
+    const matrix::SubTileSpec& spec = panel_view(idx);
+
+    // TODO this is a workaround for the deadlock problem
+    if constexpr (ForceCopy)
+      pika::dataflow(getHpExecutor<Backend::MC>(), pika::unwrapping(matrix::internal::copy_o),
+                     matrix::splitTile(mat_a.read(idx), spec), v(idx));
```

Comment:
This workaround implies copying the panel with reflectors to a temporary memory in Panel, in order to avoid the deadlock problem in matrix update.

Indeed, with the adaption of the algorithm to work with subtiles, it may end up accessing the same tile both in reading mode, for the panel reflectors data, and in write mode, because we are updating the trailing matrix.

Reply:
"The deadlock problem" being the problem that seemingly unrelated accesses on the panel (view) may in fact point to the same underlying tile?

Reply:
I'm marking this as resolved since I wasn't really expecting anything to change in this PR regarding this.

I assume this is going to be difficult to fix. Currently all that comes to mind is that something like `transform` would need to know very intimately about how tiles are accessed to be able to detect if overlapping tiles are being used in the same call, but I don't like the idea of this at all.